blocks/monitoring/aggregation.py

```diff
@@ -151,3 +151,19 @@ def get_aggregator(self):
                               (self.storage, tensor.zeros_like(self.storage))],
                           accumulation_updates=[(self.storage, self.variable)],
                           readout_variable=self.storage)
+
+
+@add_metaclass(ABCMeta)
+class MonitoredQuantity(object):
+    def __init__(self, requires=None, name=None):
+        self.requires = requires
+        self.name = name
+        pass
+
```

Comment:
I am afraid you never call the `initialize` method.

Please try change you test to cover this case, that is make sure that the test fails if reseting does not work (like now).

Also I guess you can start working on documentation. This abstract class should be well documented, the documentation should show what the inherit methods should do. Also don't forget to update docs of `DatasetEvaluator` and `DataStreamMonitoring` to reflect the fact that not only Theano variables are supported.